appIcons.js

```diff
@@ -891,6 +920,22 @@ const MyAppIconMenu = class DashToDock_MyAppIconMenu extends AppDisplay.AppIconM
                 super._redisplay();
         }
 
+        // dynamic menu
+        const items = this._getMenuItems();
+        let i = items.length;
+        if (Shell.AppSystem.get_default().lookup_app('org.gnome.Software.desktop')) {
```

Comment:
What was the reason for this special behavior?

Reply:
This is to position the quicklist menu items just below the launcher action items. There are two conditional items in the menu that appear below the launcher actions (‘Add to/Remove from Favorites’, ‘Show Details’); depending on the value of the `show-windows-preview` setting, they are added by either [upstream code](https://gitlab.gnome.org/GNOME/gnome-shell/-/blob/8745a61e4002a3c6ac737e2b51118b900dbc5d26/js/ui/appDisplay.js#L2830-2868) or a few lines above in this file, but only if certain conditions are met. To insert the quicklist section in the correct place in either branch, I need to do the same tests that upstream/the code above does, and when true, decrement the position of the quicklist section by 2 (item + separator).

Reply:
I see thanks!